### ISSUE TITLE
Creating a new directory whose name includes HTML tags results in unintended behavior

### ISSUE THREAD

**Issue by ericfranz** [OSC/ood-fileexplorer#198]
Entering invalid characters into the File Explorer’s prompt for creating a new directory results in unintended directory structure changes.

![screen 2019-03-13 at 4 33 36 PM](https://user-images.githubusercontent.com/512333/54312628-c6e3fe00-45ad-11e9-8683-9c034f4c7d1c.png)

![screen 2019-03-13 at 4 33 51 PM](https://user-images.githubusercontent.com/512333/54312640-cea3a280-45ad-11e9-81f6-6aba15a327b4.png)

**Comment by MorganRodgers** [comment]
I am unable to duplicate this behavior. When I enter the string `<script>window.alert("hacked");</script>` an error is logged by nginx, and that is it. An error should be shown to the user. Emoji are not treated as valid directory names even though they are supported by nfs4; they result in the same silent error.

Wondering if this was a more general UTF8 problem, I confirmed that Chinese characters are treated as valid input, as are accented Latin characters.

**Comment by MorganRodgers** [comment]
After a second look, I am able to duplicate with the string `<script>window.alert('hacked');</script>`; note the single quotes.

**Comment by MorganRodgers** [comment]
It appears that all the New File|Directory|Rename|Copy controls are fragile and do not permit same range of file names that the underlying file system does.

A quick fix would prevent creation of new file/directory names containing double quotes, or XML special characters.

**Comment by MorganRodgers** [comment]
I think that we decided on `wontfix` for this?

**Comment by ericfranz** [comment]
Possible problematic characters here:

- `<`
- `>`
- `'`
- `(` or `)` results in client side JavaScript error on page because `DOM.getCurrentDirName()` that returns the current directory with `(` or `)` is passed as an argument directly to `getParentDirPath` function, which is using a bad approach to resolve the parent directory string:
    ```
    > path = "/path/to/dir/name"
    '/path/to/dir/name'
    > dirName = "name"
    'name'
    > RegExp
    [Function: RegExp]
    > path.replace(RegExp(dirName + '$'), '');
    '/path/to/dir/'
    ```
    - instead we want essentially https://nodejs.org/api/path.html#path_path_dirname_path for client side use
- `/` is allowed because you can do mkdir foo/bar and it will create foo/ and then create foo/bar/

**Comment by MorganRodgers** [comment]
For what it is worth: Cloudcmd uses Handlebars which HTML-escapes-by-default unless the developer uses the triple handlebar `{{{ scary_html_here }}}`: https://handlebarsjs.com/

**Comment by MorganRodgers** [comment]
Wontfix.

**Comment by ericfranz** [comment]
This is fixed in 1.8.12, 1.7.16 and 1.6.23
